storage/innobase/log/log0log.cc

```diff
@@ -1351,82 +1321,69 @@ redo log file header.
+		/* re-fetch latest key */
+		redo_log_key *mkey = redo_log_key_mgr.load_latest_key();
+		if (mkey != NULL) {
+			space->encryption_redo_key = mkey;
+			srv_redo_log_key_version = mkey->version;
+		}
 	}
- 	/* Rotate log tablespace */
-	return(log_write_encryption(NULL, NULL, static_cast<redo_log_encrypt_enum>(srv_redo_log_encrypt)));
 }
 
 /** Check the redo log encryption is enabled or not.
 It will try to enable the redo log encryption and write the metadata to
 redo log file header. */
 void
-redo_rotate_default_key()
-{
-	fil_space_t* space = fil_space_get(SRV_LOG_SPACE_FIRST_ID);
+redo_rotate_default_key() {
+	fil_space_t *space = fil_space_get(SRV_LOG_SPACE_FIRST_ID);
 
 	if (srv_shutdown_state != SRV_SHUTDOWN_NONE) {
 		return;
 	}
 
 	/* If the redo log space is using default key, rotate it.
 	We also need the server_uuid initialized. */
-	if (space->encryption_type != Encryption::NONE
-	    && Encryption::master_key_id == ENCRYPTION_DEFAULT_MASTER_KEY_ID
-	    && !srv_read_only_mode
-	    && strlen(server_uuid) > 0
-	    && srv_redo_log_encrypt == REDO_LOG_ENCRYPT_MK) {
-			ut_ad(FSP_FLAGS_GET_ENCRYPTION(space->flags));
-
-			log_write_encryption(NULL, NULL, REDO_LOG_ENCRYPT_MK);
-	}
-
-	if (space->encryption_type != Encryption::NONE
-	    && space->encryption_key_version ==REDO_LOG_ENCRYPT_NO_VERSION 
-	    && !srv_read_only_mode
-	    && strlen(server_uuid) > 0
-	    && srv_redo_log_encrypt == REDO_LOG_ENCRYPT_RK) {
-			/* This only happens when the server uuid was just generated, so we can
-			 * save the key to the keyring */
-			if (my_key_store(PERCONA_REDO_KEY_NAME, "AES", NULL,
-						space->encryption_key, ENCRYPTION_KEY_LEN))
-			{
-				srv_redo_log_encrypt = REDO_LOG_ENCRYPT_OFF;
-				ib::error() << "Can't store redo log encryption key.";
-			}
-			uint version = 0;
-			size_t   klen = 0;
-			size_t   klen2 = 0;
-			char *redo_key_type = NULL;
-			byte *rkey = NULL;
-			unsigned char *rkey2 = NULL;
-			if (my_key_fetch(PERCONA_REDO_KEY_NAME, &redo_key_type, NULL,
-						reinterpret_cast<void**>(&rkey), &klen)) {
-				srv_redo_log_encrypt = REDO_LOG_ENCRYPT_OFF;
-				ib::error() << "Can't fetch latest redo log encryption key.";
-			}
-			const bool err = (parse_system_key(rkey, klen, &version, &rkey2, &klen2) == reinterpret_cast<uchar*>(NullS));
-			if (err) {
-				srv_redo_log_encrypt = REDO_LOG_ENCRYPT_OFF;
-				ib::error() << "Can't parse latest redo log encryption key.";
-			}
-			space->encryption_key_version = version;
-			if (!log_write_encryption(NULL, NULL, REDO_LOG_ENCRYPT_RK)) {
-				ib::error() << "Can't write redo log encryption information.";
-			}
-			if (rkey != NULL) {
-				my_free(rkey);
-			}
-			if (rkey2 != NULL) {
-				my_free(rkey2);
-			}
-			if (redo_key_type != NULL) {
-				my_free(redo_key_type);
-			}
+	if (space->encryption_type != Encryption::NONE &&
+	    Encryption::master_key_id == ENCRYPTION_DEFAULT_MASTER_KEY_ID &&
+	    !srv_read_only_mode && strlen(server_uuid) > 0 &&
```

Comment:
I think this code is no longer called from background thread (different from upstream).
Can you remove the uuid check and assert it is  > 0.